src/tests/system/tests/test_memcache.py

```diff
@@ -2,1659 +2,427 @@
+    assert client.tools.getent.group(g1.name) is not None, f"`{g1.name}` should be found while SSSD is running."
 
-    def check():
-        result = client.tools.id(2001)
-        assert result is not None, "User with id 2001 was not found using id"
-        assert result.memberof([101, 1001, 1002]), "User with id 2001 is member of incorrect groups"
-
-        result = client.tools.id(2002)
-        assert result is not None, "User with id 2002 was not found using id"
-        assert result.memberof([102, 1002]), "User with id 2002 is member of incorrect groups"
-
-        result = client.tools.id(2003)
-        assert result is not None, "User with id 2003 was not found using id"
-        assert result.memberof([103, 1002]), "User with id 2003 is member of incorrect groups"
-
-    u1 = provider.user("user1").add(uid=2001, gid=101)
-    u2 = provider.user("user2").add(uid=2002, gid=102)
-    u3 = provider.user("user3").add(uid=2003, gid=103)
-
-    provider.group("group1").add(gid=1001).add_member(u1)
-    provider.group("group2").add(gid=1002).add_members([u1, u2, u3])
-
-    client.sssd.domain["ldap_id_mapping"] = "false"
-    client.sssd.start()
-
-    check()
     client.sssd.stop()
-    check()
-
-
-@pytest.mark.importance("critical")
-@pytest.mark.cache
-@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
-def test_memcache__lookup_users_by_fully_qualified_names(client: Client, provider: GenericProvider):
-    """
-    :title: Lookup user by full name when 'use_fully_qualified_names' is 'true'
-            uses memory cache when sssd is stopped
-    :setup:
-        1. Add 'user1' and 'user2' to SSSD
-        2. In SSSD domain change 'use_fully_qualified_names' to 'true'
-        3. Start SSSD
-    :steps:
-        1. Find 'user1' and 'user2' with id(name)
-        2. Find 'user1' and 'user2' with id(name@domain)
-        3. Check that users have correct full names
-        4. Stop SSSD
-        5. Find 'user1' and 'user2' with id(name)
-        6. Find 'user1' and 'user2' with id(name@domain)
-        7. Check that users have correct full names
-    :expectedresults:
-        1. Users are not found
-        2. Users are found
-        3. Users have correct full names
-        4. SSSD is stopped
-        5. Users are not found
-        6. Users are found
-        7. Users have correct full names
-    :customerscenario: False
-    """
-
-    def check():
-        assert client.tools.id("user1") is None, "User user1 should not be found without fully qualified name"
-        assert client.tools.id("user2") is None, "User user2 should not be found without fully qualified name"
 
-        result = client.tools.id("user1@test")
-        assert result is not None, "User user1@test was not found using id"
-        assert result.user.name == "user1@test", f"User {result.user.name} has incorrect name, user1@test expected"
+    if lookup_type in ("users", "users_and_groups") and not memcache:
```

Comment:
This is ugly, it would be much better to have multiple stanalone tests.

Reply:
I did a bit of refactoring there, but not as much as you would like to. I hope we can find a middle ground.